third-party/linshare/src/main/java/org/apache/james/linshare/LinshareConfiguration.java

```diff
@@ -30,67 +30,82 @@
 
 public class LinshareConfiguration {
 
-    public static final String URL = "blob.export.linshare.url";
-    public static final String TOKEN = "blob.export.linshare.token";
+    public static final String URL_PROPERTY = "blob.export.linshare.url";
+    public static final String USER_NAME_PROPERTY = "blob.export.linshare.user.name";
+    public static final String PASSWORD_PROPERTY = "blob.export.linshare.user.password";
 
     public static class Builder {
         @FunctionalInterface
         public interface RequireUrl {
-            RequireAuthorizationToken url(URL url);
+            RequireBasicAuthorization url(URL url);
 
-            default RequireAuthorizationToken urlAsString(String url) throws MalformedURLException {
+            default RequireBasicAuthorization urlAsString(String url) throws MalformedURLException {
                 return url(new URL(url));
             }
         }
 
-        public interface RequireAuthorizationToken {
-            ReadyToBuild authorizationToken(AuthorizationToken token);
+        public interface RequireBasicAuthorization {
+            ReadyToBuild basicAuthorization(String username, String password);
         }
 
         public static class ReadyToBuild {
             private final URL url;
-            private final AuthorizationToken token;
+            private final String username;
+            private final String password;
 
-            ReadyToBuild(URL url, AuthorizationToken token) {
+            ReadyToBuild(URL url, String username, String password) {
                 this.url = url;
-                this.token = token;
+                this.username = username;
+                this.password = password;
             }
 
             public LinshareConfiguration build() {
-                return new LinshareConfiguration(url, token);
+                return new LinshareConfiguration(url, username, password);
             }
         }
     }
 
     public static Builder.RequireUrl builder() {
-        return url -> credential -> new Builder.ReadyToBuild(url, credential);
+        return url -> (username, password) -> new Builder.ReadyToBuild(url, username, password);
     }
 
     public static LinshareConfiguration from(Configuration configuration) throws MalformedURLException {
         return builder()
-            .urlAsString(configuration.getString(URL, null))
-            .authorizationToken(new AuthorizationToken(configuration.getString(TOKEN, null)))
+            .urlAsString(configuration.getString(URL_PROPERTY, null))
+            .basicAuthorization(
+                configuration.getString(USER_NAME_PROPERTY, null),
+                configuration.getString(PASSWORD_PROPERTY, null))
             .build();
     }
 
     private final URL url;
-    private final AuthorizationToken token;
+    private final String username;
```

Comment:
Should use strong typing `Username` ?

Reply:
Field **username** now used for both **user** and **technicalAccount**.
For now, **technicalAccount** will do **basicAuth** with  **UUID/password** so i guess we can use normal type as **String**?

Reply:
If I understood correctly, a refactoring of that will happen in an other PR right? Don't forget about it then please :)

Reply:
Yes, is will be a next PR about it.